R/utils.R

```diff
@@ -510,3 +510,29 @@ process_theme <- function(theme) {
 
     return(getFromNamespace(parts[2], parts[1]))
 }
+
+
+#' check_palette
+#' Check if the palette and palcolor can be properly used
+#' @param palette palette or palcolor
+#' @param datas_name name of the split data
+#' @keywords internal
+#' @return named list containing palette or color name
+check_palette <- function(palette, datas_name) {
+    if (length(palette) > 1) {
+		if (!is.list(palette)) {
+			stopifnot("palette (palcolor) should be named if provided as a vector." = !is.null(names(palette)))
+			palette <- split(unname(palette), names(palette))
+		} else {
+			stopifnot("palette (palcolor) should be named if provided as a list." = !is.null(names(palette)))
+		}
+        stopifnot("split_by and palette (palcolor) length mismatches." = length(palette) == length(datas_name))
+    } else {
+    	if (!is.null(palette)) {
+	        palette <- rep(palette, length(datas_name))
+	        names(palette) <- datas_name
+			palette <- split(unname(palette), names(palette))	
+    	}
+    }
+    return(palette)
+}
```

Comment:
We could either separate them (i.e. `check_palette` and `check_palcolor`), or implement them in the same function but handle them separately:

```r
check_palette <- function(palette, palcolor, datas_name) {
  # have them handled separately and message the user exactly which one is wrong when it is
  return(list(palette = palette, palcolor = palcolor))
}

## use it
valid_palette <- check(palette, palcolor, datas_name)
palette <- valid_palette$palette
palcolor <- valid_palette$palcolor
```

Have a separate function to check palcolor might have a better readability.